File: services/inspection-service/src/main/java/com/shanghaiwind/inspection/entity/Project.java
package com.shanghaiwind.inspection.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 项目实体
 */
@Data
@TableName("biz_project")
public class Project {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 项目名称
     */
    private String name;

    /**
     * 项目编码
     */
    private String code;

    /**
     * 业主
     */
    private String owner;

    /**
     * 项目位置
     */
    private String location;

    /**
     * 纬度
     */
    private Double lat;

    /**
     * 经度
     */
    private Double lng;

    /**
     * 状态: ACTIVE, COMPLETED, ARCHIVED
     */
    private String status;

    /**
     * 项目经理ID
     */
    private Long managerId;

    /**
     * 描述
     */
    private String description;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: apps/web/src/api/project.ts
import request from '@/utils/request'

export interface Project {
  id: number
  name: string
  code: string
  owner: string
  location: string
  lat: number
  lng: number
  status: string
  managerId: number
  description: string
  createdAt: string
  updatedAt: string
}

export interface ProjectDTO {
  name: string
  code?: string
  owner?: string
  location?: string
  lat?: number
  lng?: number
  managerId?: number
  description?: string
}

// 分页查询项目
export function getProjectPage(params: {
  pageNum: number
  pageSize: number
  keyword?: string
  status?: string
}) {
  return request.get<any, { records: Project[]; total: number }>('/inspection/projects', { params })
}

// 获取所有项目
export function getAllProjects() {
  return request.get<any, Project[]>('/inspection/projects/all')
}

// 获取项目详情
export function getProjectDetail(id: number) {
  return request.get<any, Project>(`/inspection/projects/${id}`)
}

// 创建项目
export function createProject(data: ProjectDTO) {
  return request.post<any, Project>('/inspection/projects', data)
}

// 更新项目
export function updateProject(id: number, data: ProjectDTO) {
  return request.put<any, Project>(`/inspection/projects/${id}`, data)
}

// 删除项目
export function deleteProject(id: number) {
  return request.delete(`/inspection/projects/${id}`)
}
